### ISSUE TITLE
Allow easy configuration of idleness monitor in shutdown-idle

### ISSUE THREAD

**Issue by unman** [QubesOS/qubes-issues#8125]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### The problem you're addressing (if any)
Currently the timeout for the idle monitor is hard coded in idleness_monitor.py
It's trivial to change this from the default of 15*60, but any changes are overriden when the package is updated.

### The solution you'd like
Allow user to set timeout in (e.g.) /rw/config/idle_timeout.
A value set here will override the default value.

### The value to a user, and who that user might be
Allow customisation without fear of overriding on package update.

**Comment by 3hhh** [comment]
The idleness monitor assumes the executing VM to be trusted. This is a rather odd choice for Qubes OS and makes any time spent working on it rather misspent IMHO.

**Comment by marmarek** [comment]
> The idleness monitor assumes the executing VM to be trusted. This is a rather odd choice for Qubes OS and makes any time spent working on it rather misspent IMHO.

Can you elaborate? The purpose of this thing is to shutdown the qube when it isn't doing anything (note the important distinction from "user isn't interacting with the qube"). If you worry about malicious qube preventing the shutdown, then instead of messing with the tool, it could simply pretend the qube is "doing something".

**Comment by 3hhh** [comment]
> > The idleness monitor assumes the executing VM to be trusted. This is a rather odd choice for Qubes OS and makes any time spent working on it rather misspent IMHO.
> 
> Can you elaborate? The purpose of this thing is to shutdown the qube when it isn't doing anything (note the important distinction from "user isn't interacting with the qube"). If you worry about malicious qube preventing the shutdown, then instead of messing with the tool, it could simply pretend the qube is "doing something".

Yes, with that definition it does what it's supposed to do and the implementation inside the target qube makes some sense. However that definition isn't what most users incl. me would expect from a security-focused OS.

I'd rather expect a _reliable_ and _unstoppable_ shutdown of a configurable set of qubes when either the user is not doing any input for X minutes or a qube didn't spawn any window for Y minutes. The latter can be faked, but is rather obvious to spot once a user becomes active again. Maybe there are also better methods to detect a "VM idle" state without trusting the VM too much. Of course such code would have to run inside the admin VM, i.e. mostly dom0.
(Btw I implemented it that way [some years ago](https://github.com/3hhh/qidle).)

**Comment by andrewdavidwong** [comment]
Allowing the idle timeout to be user-configurable was originally requested and discussed in #5298. I'll repost @marmarta's [comment](https://github.com/QubesOS/qubes-issues/issues/5298#issuecomment-670733947) from that issue for the sake of context and historical continuity:

> Okay, so, because making the time period really configurable is a bit of a pain and I'd hate for the better to be the enemy of the good, I'm adding a "hardcoded 15 minutes" one to VM settings and if I have more time some day I'll make it more configurable. Alternatively, patches welcome :)
